code/src/components/LikeButton.js

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+
+export const LikeButton = (props) => {
+  const [like, setLike] = useState(false);
+  const [hearts, setHearts] = useState(props.hearts)
+  const likeButtonToggle = like ? ' on ' : '';
+  /* will post the like, keep count of likes */
+  const postLike = () => {
+    return fetch(`https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts/${props.id}/like`, {
+      method: 'POST'
+    })
+  }
+  const toggleLike = () => {
+    postLike().then((response) => response.json()).then((data) => {
+      setHearts(data.hearts)
+      if (!like) {
+        setLike(true);
+      }
+    })
+  }
+  return (
+    <div>
+      <button
+        aria-label="Like-button"
+        type="button"
+        className={`likeButton${likeButtonToggle}`}
+        disabled={like}
+        onClick={toggleLike}>
+        ❤️
+      </button>
+      <span>x {hearts}</span>
+    </div>
+  )
+}
```

Comment:
nice you figured out how to limit the clicks on the hearts i missed that in mine